TemperatureParameter.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+
+public class TemperatureParameter : IBatteryParameter
+{
+    public float limitMin = 0;
+    public float limitMax = 45;
+
+    public bool EarlyWarningEnabled { get; set; } = true;
+    public float EarlyWarningTolerencePercentage { get; set; } = 5;
+
+    public TemperatureParameter(bool enableEarlyWarning, float earlyWarningTolerence)
+    {
+        EarlyWarningEnabled = enableEarlyWarning;
+        EarlyWarningTolerencePercentage = earlyWarningTolerence;
+    }
+
+    public bool Validate(float value, Action<string> printCallback)
+    {
+        if (MathUtils.IsValueInRange(value, limitMin, limitMax))
+        {
+            CheckEarlyWarning(value, printCallback);
+            return true;
+        }
+
+        printCallback(Localization.GetInSelectedLanguage(Localization.Phrases.Temperature_OutOfRange));
+        return false;
+    }
+
+    public void CheckEarlyWarning(float value, Action<string> printCallback)
+    {
+        if (EarlyWarningEnabled)
+        {
+            float valueRange = limitMax - limitMin;

```

Comment:
there can be an improvement in reducing the number of calculations. Tolerance value remains constant for one program execution. But in this case it is calculated every time.
Please think if this can be optimized.